rubin_sim/maf/metrics/surfbMetric.py

```diff
@@ -0,0 +1,180 @@
+import numpy as np
+from .baseMetric import BaseMetric
+from rubin_sim.photUtils import Bandpass, PhotometricParameters
+import rubin_sim.utils as rsUtils
+from rubin_sim.data import get_data_dir
+import os
+
+__all__ = ["SurfaceBrightLimitMetric"]
+
+
+def surface_brightness_limit_approx(
+    zp, k, airmass, mu_sky, rn=8.8, pixscale=0.2, nsigma=3.0, t_exp=30.0, tot_area=100.0
+):
+    """Compute surface brightness limit in 3 limiting cases, return the brightest.
+
+    Algerbra worked out in this notebook: 
+    https://github.com/yoachim/21_Scratch/blob/main/surfb_metric/surface_brightness_limit.ipynb
+
+    Parameters
+    ---------
+    zp : float
+        Telescope zeropoint (mags)
+    k : float
+        Atmospheric extinction term
+    airmass : float
+        Airmass
+    mu_sky : float
+        Surface brightness of the sky (mag/sq arcsec)
+    rn : float (8.8)
+        Readnoise in electrons
+    pixscale : float (0.2)
+        Arcseconds per pixel
+    nsigma : float (3)
+        The SNR to demand
+    t_exp : float (30.)
+        Exposure time (seconds)
+    tot_area : float (100)
+        The total area measuring over (sq arcsec)
+
+    Returns
+    -------
+    surface brightness limit in mags/sq arcsec
+    aka the surface brightness that reaches SNR=nsigma when measured over tot_area.
+    """
+
+    n_pix = tot_area / pixscale
+
+    A_pix = pixscale ** 2
+
+    # Sky limited case
+    mu_sky_lim = (
+        -1.25 * np.log10(nsigma ** 2 / (A_pix * t_exp * n_pix))
+        + 0.5 * mu_sky
+        + 0.5 * zp
+        - k * airmass
+    )
+
+    # Source limited case
+    # XXX--double check this algerbra. Pretty sure it's right now.
+    mu_source_lim = (
+        -1.25 * np.log10(nsigma)
+        + 1.25 / 2.0 * np.log10(n_pix * A_pix * t_exp)
+        + zp
+        - k * airmass
+    )
+
+    # Readnoise limited case
+    mu_rn_lim = (
+        -2.5 * np.log10(nsigma * rn / (t_exp * A_pix * n_pix ** 0.5)) + zp - k * airmass
+    )
+
+    result = np.vstack([mu_sky_lim, mu_source_lim, mu_rn_lim])
+    result = np.min(result, axis=0)
```

Comment:
This might be off if no single origin for the noise dominates, e.g. if the readnoise is about the same as the sky noise and source noise is negligable, this will underestimate the noise by a factor of sqrt(2), so overestimate the depth by 2.5*log10(sqrt(2))=0.38, or more if all 3 are comparable.

Reply:
Yup. Added a check so it'll throw a warning if it's doing a potentially lousy approximation.

Reply:
Good enough, unless we later find the warning gets thrown regularly. If we find that it is, we may need to combine them properly, but that can be a separate issue, not part of this commit.